python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py

```diff
@@ -0,0 +1,132 @@
+import json
+import os
+import subprocess
+from dataclasses import dataclass
+from typing import Any, Dict, Iterator, List, Sequence
+
+from dagster import _check as check
+
+
+@dataclass
+class DbtCliEventV2:
+    """Represents a dbt CLI event."""
+
+    event: Dict[str, Any]
+
+    @classmethod
+    def from_log(cls, log: str) -> "DbtCliEventV2":
+        """Parse an event according to https://docs.getdbt.com/reference/events-logging#structured-logging.
+
+        We assume that the log format is json.
+        """
+        event: Dict[str, Any] = json.loads(log)
+
+        return cls(event=event)
+
+
+class DbtCliTask:
+    def __init__(self, process: subprocess.Popen) -> None:
+        self._process = process
+
+    def wait(self) -> Sequence[DbtCliEventV2]:
+        """Wait for the dbt CLI process to complete and return the events.
+
+        Returns:
+            Sequence[DbtCliEventV2]: A sequence of events from the dbt CLI process.
+        """
+        return list(self.stream())
+
+    def stream(self) -> Iterator[DbtCliEventV2]:
+        """Stream the events from the dbt CLI process.
+
+        Returns:
+            Iterator[DbtCliEventV2]: An iterator of events from the dbt CLI process.
+        """
+        for raw_line in self._process.stdout or []:
+            log: str = raw_line.decode().strip()
+            event = DbtCliEventV2.from_log(log=log)
+
+            yield event
+
+        # TODO: handle the return codes here!
+        # https://docs.getdbt.com/reference/exit-codes
+        return_code = self._process.wait()
+
+        return return_code
+
+
+class DbtClientV2:
+    def __init__(self) -> None:
+        # TODO: We should probably check that the dbt-core version here is above 1.4.0.
+        check.invariant(
```

Comment:
commented along these lines on the notion doc, but I don't think we should rely solely on environment variables here, as this is generally going to bewilder users who aren't familiar w/ setting env vars, and likely cause headaches when people try to deploy their local code to prod.

instead, we can have a permissive config schema -- any flags set therein would be passed into CLI invocations (any of these global config env vars can also be passed in as cli flags). we can do some proactive error checking for the user (like if they don't configure a `project_dir`, and their `DBT_PROJECT_DIR` env var is not set, then we can error early), but in general I don't think we should requires these to be set for the integration to work.